src/System.Net.Requests/src/System/Net/FtpWebRequest.cs

```diff
@@ -0,0 +1,2544 @@
+        ///         Effects Streams returned from GetResponse().GetResponseStream() (AND) GetRequestStream().
+        ///         Default is 5 mins.
+        ///    </para>
+        /// </devdoc>
+        public int ReadWriteTimeout {
+            get {
+                return m_ReadWriteTimeout;
+            }
+            set {
+                if (m_GetResponseStarted) {
+                    throw new InvalidOperationException(SR.net_reqsubmitted);
+                }
+                if (value<=0 && value!=System.Threading.Timeout.Infinite) {
+                    throw new ArgumentOutOfRangeException("value", SR.net_io_timeout_use_gt_zero);
+                }
+                m_ReadWriteTimeout = value;
+            }
+        }
+
+        /// <summary>
+        /// <para>Used to specify what offset we will read at</para>
+        /// </summary>
+        public long ContentOffset {
+            get {
+                return m_ContentOffset;
+            }
+            set {
+                if (InUse) {
+                    throw new InvalidOperationException(SR.net_reqsubmitted);
+                }
+                if (value<0) {
+                    throw new ArgumentOutOfRangeException("value");
+                }
+                m_ContentOffset = value;
+            }
+        }
+
+
+
+        /// <summary>
+        /// <para>Gets or sets the data size of to-be uploaded data</para>
+        /// </summary>
+        public override long ContentLength {
+            get {
+                return m_ContentLength;
+            }
+            set {
+                m_ContentLength = value;
+            }
+        }
+
+        /// <summary>
+        /// <para>Uses an HTTP proxy if needed to send FTP request</para>
+        /// </summary>
+        public override IWebProxy Proxy {
+            get {
+                return m_Proxy;
+            }
+            set {
+                if (InUse) {
+                    throw new InvalidOperationException(SR.net_reqsubmitted);
+                }
+                m_ProxyUserSet = true;
+                m_Proxy = value;
+                m_ServicePoint = null;
+                ServicePoint refreshIt = ServicePoint ;
+            }
+        }
+
+        /// <devdoc>
+        /// <para>Allows private ConnectionPool(s) to be used</para>
+        /// </devdoc>
+        public override string ConnectionGroupName {
+            get {
+                return m_ConnectionGroupName;
+            }
+            set {
+                if (InUse) {
+                    throw new InvalidOperationException(SR.net_reqsubmitted);
+                }
+                m_ConnectionGroupName = value;
+            }
+        }
+
+        /// <devdoc>
+        /// <para>Generates a service point for this request, and allows setting of Connection settings</para>
+        /// </devdoc>
+        public ServicePoint ServicePoint {
+            get {
```

Comment:
We are.  The only reason it was removed from Mail was to avoid the dependency from System.Net.Mail to System.Net.Requests.